crash-handler/src/main/java/org/odk/collect/crash_handler/CrashHandler.kt

```diff
@@ -0,0 +1,88 @@
+package org.odk.collect.crash_handler
+
+import android.content.Context
+import org.odk.collect.androidshared.data.getState
+import kotlin.system.exitProcess
+
+class CrashHandler(private val processKiller: Runnable = Runnable { exitProcess(0) }) {
+
+    private var conditionFailure: String? = null
+
+    fun registerCrash(context: Context, crash: Throwable) {
+        getPreferences(context).edit().putString(KEY_CRASH, crash.message ?: "").apply()
+    }
+
+    fun checkConditions(runnable: Runnable): Boolean {
+        return try {
+            runnable.run()
+            true
+        } catch (t: Throwable) {
+            conditionFailure = t.message ?: ""
+            false
+        }
+    }
+
+    fun hasCrashed(context: Context): Boolean {
+        return getPreferences(context).contains(KEY_CRASH) || conditionFailure != null
+    }
+
+    @JvmOverloads
+    fun getCrashView(context: Context, onErrorDismissed: Runnable? = null): CrashView? {
+        val preferences = getPreferences(context)
+
+        return if (conditionFailure != null) {
+            val crashMessage = conditionFailure
+
+            CrashView(context).also {
+                it.setCrash(context.getString(R.string.cant_start_app), crashMessage) {
+                    processKiller.run()
+                }
+            }
+        } else if (preferences.contains(KEY_CRASH)) {
+            val crashMessage = preferences.getString(KEY_CRASH, null)
+
+            CrashView(context).also {
+                it.setCrash(context.getString(R.string.crash_last_run), crashMessage) {
+                    preferences.edit().remove(KEY_CRASH).apply()
+                    onErrorDismissed?.run()
+                }
+            }
+        } else {
+            null
+        }
+    }
+
+    private fun getPreferences(context: Context) =
+        context.getSharedPreferences("crash_handler", Context.MODE_PRIVATE)
+
+    companion object {
+
+        private const val KEY_CRASH = "crash"
+
+        @JvmStatic
+        fun install(context: Context): CrashHandler {
+            val crashHandler = CrashHandler()
```

Comment:
I think it would make more sense to pass `crashHandler` to app state immediately here:
```
           val crashHandler = CrashHandler().also {
                context.getState().set("crash_handler", it)
            }
```